### PR TITLE
fix(Core/Spells): Ensures that procs triggered by another aura are pr…

### LINKED ISSUE #4720: Shaman Flurry talent on consecutive crits [$10]
Originally reported https://github.com/chromiecraft/chromiecraft/issues/122

##### ISSUE: FACTION SIDE:

 Alliance
 Horde

##### CONTENT PHASE:

20-29

##### SMALL DESCRIPTION:

Spell: [Flurry](https://wowgaming.altervista.org/aowow/?spell=16284)

##### EXPECTED BLIZZLIKE BEHAVIOUR:

After dealing a melee critical hit, this talent should give 3 charges which buff you with 30% increased attack speed and are each expended by the following 3 melee swings.
When charges from a previous crit remain and a new critical hit is dealt, the new crit should set the charges to three, they don't stack.
https://www.youtube.com/watch?v=jvTAzjotu60 at 5:54
Gets a crit on the hunter, swings once at the druid expending one charge, next swing on the druid is a crit and the stacks go from 2 to 3.
I could not find documented changes to how the stacks work between vanilla final patch and last patch of wotlk.

##### CURRENT BEHAVIOUR:

Works normally on the first critical hit.
When having 1 or more charges of a previous critical remaining, if another critical hit is dealt, the buff duration gets reset but the charges will be set to only 2.

##### STEPS TO REPRODUCE THE PROBLEM:

Step 1 Spec the flurry talent in the Enhancement shaman tree
Step 2 Hit a creature until getting a critical
Step 3 Get a crit with one of the three swings following the proc from the previous step

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/c2f274e06dcf85c6ccc0ab282dd6f23e8ce49c8f

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

<bountysource-plugin>

---
There is a **[$10 open bounty](https://www.bountysource.com/issues/97032832-shaman-flurry-talent-on-consecutive-crits?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** on this issue. Add to the bounty at [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by UltraNix**
…occessed in the first place.

Fixed #4720.

## Changes Proposed:
-  Aura procs triggered by another proc aura should be proccessed in the first place.
-  

## Issues Addressed:
- Closes #4720.

## Tests Performed:

- Tested ingame with Rogue's Flurry.
- 

## Known Issues and TODO List:

- [ ]
- [ ] 

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by UltraNix** [comment]
@Stifler82 Please, test it.

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -5855,6 +5855,7 @@ void Unit::SendPeriodicAuraLog(SpellPeriodicAuraLogInfo* pInfo)
                     damage = 0;
                 }
 
+
```
Comment (Kitzunu) [inline src/server/game/Entities/Unit/Unit.cpp:5858]: remove pls

**Comment by UltraNix** [comment]
@Kitzunu Done.

**Comment by Stifler82** [comment]
@UltraNix 
If i read good the Fix, this isn't a specifici Fix, for the specific issue, right?
This is a Fix that make work like it intended all Spells that work like Flurry, right?(which should be all spells with Stacks) right?
so this is a "Generic Fix" right?

**Comment by UltraNix** [comment]
This is a generic fix for spells that triggers another spell and both can proc at the same time, as e.g. Flurry. In that case, the triggered spell should be handled first in order to properly "eat" stacks/charges before the "main" spell proc it again.

**Comment by Stifler82** [comment]
> This is a generic fix for spells that triggers another spell and both can proc at the same time, as e.g. Flurry. In that case, the triggered spell should be handled first in order to properly "eat" stacks/charges before the "main" spell proc it again.

@UltraNix 
Ok, maybe I explained it badly, (I apologize), but yes, I understood well then..
But tell me, and if there is a spell that works in Stack, (BUT!) It works slightly different and it doesn't have to reach (The maximum number of Stacks) with this generic fix don't we bugger that spell?
(now I don't have a practical example at hand, because right now they don't come to mind)..

**Comment by UltraNix** [comment]
So you have to tell me that example so I can look at it. For now, I don't see any.

**Comment by Stifler82** [comment]
> So you have to tell me that example so I can look at it. For now, I don't see any.

Warning: I'm not criticizing your fix, on the contrary, I think it's a Smart Fix.
But I wondered, and if there is a spell that works different, I would not want to fix a spell, and then we bugged others 10

**Comment by UltraNix** [comment]
Then again: test it and give me feedback about bugged spell. I create fixes, not bugs (not on purpose). I always check any influnce of my fixes on the core. Wondering or maybe "theorizing" about potential bugs is a waste of time. Just test it.

**Comment by Stifler82** [comment]
> Then again: test it and give me feedback about bugged spell. I create fixes, not bugs (not on purpose). I always check any influnce of my fixes on the core. Wondering or maybe "theorizing" about potential bugs is a waste of time. Just test it.

@UltraNix 
Ok, i'm going to test it, and I let you know..

**Comment by Stifler82** [comment]
@UltraNix 
Ok, tested, the Fix works perfectly.
For me it can be merged (As for the Flurry talent).
Then, as said before, I don't know if this **"Generic" Fix** will lead to side effects, (in other Spells / talents).
**Video Test:** [Flurry Fixed](https://youtu.be/1Dzn_LvKf0I)

**Comment by Stifler82** [comment]
@UltraNix 
Now, having solved this problem, there is another question, or perhaps better to say, **"Dilemma"**, which is not directly related to this fix, but is related to talent.

it's all explained here: [Flurry Dilemma](https://youtu.be/uYI78Y4b3sk)
more than explained it is "Illustrated" here...

**Comment by UltraNix** [comment]
`Wrath of the Lich King Patch 3.1.0 (2009-04-14): Speed increase changed to 5/10/15/20/25% (down from 10/15/20/25/30%)`
I don't see any bug here. Blizzard many times forgot to update spell tooltips.  It should be 5% per talent point and this value comes directly from Blizz files.

**Comment by Stifler82** [comment]
@UltraNix 
What I showed you, is NOT the TOOLTIP!
what I have shown you, are the TALENTS that are WRONG!

**Comment by UltraNix** [comment]
Explain to me where is the bug? Maybe I misunderstood you.

**Comment by FrancescoBorzi** [comment]
@Stifler82 can this PR be merged?

**Comment by Stifler82** [comment]
@UltraNix 
Video: [Flurry 3](https://youtu.be/2gcvkZiHL44)

**Comment by UltraNix** [comment]
@Stifler82 Now I got it :)

**Comment by UltraNix** [comment]
https://wowpedia.fandom.com/wiki/Flurry_(warrior_ability)
https://wowpedia.fandom.com/wiki/Flurry_(shaman_ability) 
For warriors values changes with TBC patch and for shaman in wotlk? Weird.

**Comment by Stifler82** [comment]
> @Stifler82 Now I got it :)
> So the talent's tooltip says 30%, aura with 30% is applied and Wowpedia/wow patches says it should be 25%. Something is wrong here.

No, wowpedia But Blizzard's WOTLK Patch Notes..

https://www.toxicgaming.us/wow/wrath-patch-notes/

**Comment by Stifler82** [comment]
> https://wowpedia.fandom.com/wiki/Flurry_(warrior_ability)
> https://wowpedia.fandom.com/wiki/Flurry_(shaman_ability)
> For warriors values changes with TBC patch and for shaman in wotlk? Weird.

I don't understand the comparison you are making, one is the talent of the shaman, the other is the talent of the warrior.

They have the same name, the same function, but they are 2 different things, but above all 2 different IDs.

**Comment by UltraNix** [comment]
Because they are the same effect and the same changes has been done.

**Comment by Stifler82** [comment]
> Because they are the same effect and the same changes has been done.

They simply made an arbitrary choice, without any particular logic, they first changed the talent of the warrior, then maybe it fit well for them as they changed the talent of the warrior in TBC and later in WOTLK they did the same with the talent of the Shaman, so what's Weird?

**Comment by UltraNix** [comment]
Maybe you are right that they decided to change that to shaman after seeing results with warrior.

**Comment by Stifler82** [comment]
I made another video for you where I clarify an important point that I'm not sure is understood in the last video.
**Video:** [Flurry 4](https://youtu.be/oCn2cUZm44w)

**Comment by UltraNix** [comment]
@Stifler82 Test it now. Changed values serverside. But ingame tooltips are hardcoded in client, so I can't change them.

**Comment by Stifler82** [comment]
> @Stifler82 Test it now. Changed values serverside. But ingame tooltips are hardcoded in client, so I can't change them.

i can fix it, don't worry👍

**Review by Stifler82 — CHANGES_REQUESTED** [review]
@UltraNix 
I don't know, maybe I explained myself badly..
You put the wrong ID's spell..

you have to correct it in this way:
`ApplySpellFix({ 12319, 12971, 12972, 12973, 12974 }, [](SpellEntry* spellInfo)`

**Comment by UltraNix** [comment]
Nope. `12319, 12971, 12972, 12973, 12974` are proc auras that trigger the ones I corrected.

**Comment by Stifler82** [comment]
**Do not merge this fix yet!!!**
-
there have been errors both from Me and from UltraNix, now I am too tired and I cannot explain, tomorrow when I wake up, (since I have been awake for 48h+) we will discuss it better.

**Comment by UltraNix** [comment]
If I understand correctly:
The problem was with Shaman's Flurry and its ranks -  was 6/12/18/24/30%, instead of 5/10/15/20/25%, just like the version for warriors. So I changed values and now they are same as Warrior's flurry - 5/10/15/20/25%.
`12319, 12971, 12972, 12973, 12974` - these IDs are warriors' Flurry talents, so I needed the Shamans' ones: `16256, 16281, 16282, 16283, 16284`. These are only proc triggered auras, so I fixed the triggered auras - `16257, 16277, 16278, 16279, 16280`.
To verify it, test ingame and once the Flurry proc - type `.list auras` and see the Flurry proced aura value (or look at weapon speed in char sheet).

**Comment by Stifler82** [comment]
So @UltraNix, we made a bit of mess, but I know how to fix everything, all you have to do is:
Cancel this fix: `src/server/game/Spells/SpellMgr.cpp`
And just leave this fix: `src/server/game/Entities/Unit/Unit.cpp`

In a while I will post here the fix to be associated with this fix: `src/server/game/Spells/SpellMgr.cpp`
ok?

**Comment by UltraNix** [comment]
But changes in `SpellMgr.cpp` are right or no? If so, what is wrong there?

**Comment by Stifler82** [comment]
> But changes in `SpellMgr.cpp` are right or no?

- Yes `SpellMgr.cpp`

>If so, what is wrong there?

- I'm working on it right now, and I'll explain you better when I'll finish it for make it clear to you too, i can't explain it now without a practice demonstration.

**Comment by Stifler82** [comment]
@UltraNix 
This is the fix you need to add (in addition to yours), with this fix all Talent Ranks will work at the correct percentages.

```
DELETE FROM `spell_dbc` WHERE (`ID` IN (16256, 16257, 16277, 16278, 16279, 16280, 16281, 16282, 16283, 16284));
INSERT INTO `spell_dbc` VALUES
(16256, 0, 0, 0, 208, 0, 0, 67108864, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 69652, 100, 0, 0, 0, 1, 21, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, -1, 0, 0, 6, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 42, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 16257, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 108, 0, 50, 'Flurry','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 'Rank 1','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 'Increases your attack speed by $16257s1% for your next 3 swings after dealing a critical strike.','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190,'','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0),
(16257, 0, 0, 0, 0, 0, 0, 524288, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 69652, 100, 3, 0, 0, 1, 8, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, -1, 0, 0, 6, 0, 0, 1, 0, 0, 0, 0, 0, 4, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 138, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2759, 0, 108, 0, 50, 'Flurry','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 'Rank 1','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 'Increases your attack speed by $16257s1% for your next 3 swings after dealing a critical strike.','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 'Attack speed increased by $16257s1%.','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 0, 0, 0, 0, 11, 0, 512, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0),
(16277, 0, 0, 0, 0, 0, 0, 524288, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 69652, 100, 3, 0, 0, 1, 8, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, -1, 0, 0, 6, 0, 0, 1, 0, 0, 0, 0, 0, 9, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 138, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2759, 0, 108, 0, 50, 'Flurry','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 'Rank 2','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 'Increases your attack speed by $16277s1% for your next 3 swings after dealing a critical strike.','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 'Attack speed increased by $16277s1%.','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 0, 0, 0, 0, 11, 0, 512, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0),
(16278, 0, 0, 0, 0, 0, 0, 524288, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 69652, 100, 3, 0, 0, 1, 8, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, -1, 0, 0, 6, 0, 0, 1, 0, 0, 0, 0, 0, 14, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 138, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2759, 0, 108, 0, 50, 'Flurry','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 'Rank 3','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 'Increases your attack speed by $16278s1% for your next 3 swings after dealing a critical strike.','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 'Attack speed increased by $16278s1%.','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 0, 0, 0, 0, 11, 0, 512, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0),
(16279, 0, 0, 0, 0, 0, 0, 524288, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 69652, 100, 3, 0, 0, 1, 8, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, -1, 0, 0, 6, 0, 0, 1, 0, 0, 0, 0, 0, 19, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 138, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2759, 0, 108, 0, 50, 'Flurry','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 'Rank 4','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 'Increases your attack speed by $16279s1% for your next 3 swings after dealing a critical strike.','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 'Attack speed increased by $16279s1%.','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 0, 0, 0, 0, 11, 0, 512, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0),
(16280, 0, 0, 0, 0, 0, 0, 524288, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 69652, 100, 3, 0, 0, 1, 8, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, -1, 0, 0, 6, 0, 0, 1, 0, 0, 0, 0, 0, 24, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 138, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2759, 0, 108, 0, 50, 'Flurry','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 'Rank 5','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 'Increases your attack speed by $16280s1% for your next 3 swings after dealing a critical strike.','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 'Attack speed increased by $16280s1%.','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 0, 0, 0, 0, 11, 0, 512, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0),
(16281, 0, 0, 0, 208, 0, 0, 67108864, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 69652, 100, 0, 0, 0, 1, 21, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, -1, 0, 0, 6, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 42, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 16277, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 108, 0, 50, 'Flurry','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 'Rank 2','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 'Increases your attack speed by $16277s1% for your next 3 swings after dealing a critical strike.','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190,'','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0),
(16282, 0, 0, 0, 208, 0, 0, 67108864, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 69652, 100, 0, 0, 0, 1, 21, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, -1, 0, 0, 6, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 42, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 16278, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 108, 0, 50, 'Flurry','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 'Rank 3','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 'Increases your attack speed by $16278s1% for your next 3 swings after dealing a critical strike.','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190,'','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0),
(16283, 0, 0, 0, 208, 0, 0, 67108864, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 69652, 100, 0, 0, 0, 1, 21, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, -1, 0, 0, 6, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 42, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 16279, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 108, 0, 50, 'Flurry','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 'Rank 4','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 'Increases your attack speed by $16279s1% for your next 3 swings after dealing a critical strike.','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190,'','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0),
(16284, 0, 0, 0, 208, 0, 0, 67108864, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 69652, 100, 0, 0, 0, 1, 21, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, -1, 0, 0, 6, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 42, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 16280, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 108, 0, 50, 'Flurry','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 'Rank 5','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 'Increases your attack speed by $16280s1% for your next 3 swings after dealing a critical strike.','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190,'','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0);
```

**Comment by UltraNix** [comment]
Are you overriding existing spells? What differences between these entries and from dbc?

**Comment by Stifler82** [comment]
> Are you overriding existing spells? What differences between these entries and from dbc?

Login on Discord

**Comment by UltraNix** [comment]
@FrancescoBorzi This can be merged as it fixed the addressed issue.
@Stifler82 is going to fix another problem with this spell out of scope this PR.

**Comment by Stifler82** [comment]
it can be merged.
@FrancescoBorzi @Helias

**Review by FrancescoBorzi — APPROVED** [review]

**Review by Stifler82 — APPROVED** [review]
Tested

**Pull request merged**